Repository/AbstractPhpcrRepository.php

```diff
@@ -147,6 +148,15 @@ public function add($path, $resource)
     }
 
     /**
+     * {@inheritdoc}
+     */
+    public function reorder($sourcePath, $position)
+    {
+        Assert::greaterThanEq($position, 0, 'Reorder position cannot be negative, got: %s');
+        $this->reorderNode($sourcePath, $position);
```

Comment:
please define an abstract method reorderNode as all child classes need that method